src/api/readCpmProduct/src/v1/steps.py

```diff
@@ -0,0 +1,33 @@
+from aws_lambda_powertools.utilities.data_classes import APIGatewayProxyEvent
+from domain.repository.cpm_product_repository.v3 import CpmProductRepository
+from domain.request_models.v1 import CpmProductPathParams
+from domain.response.validation_errors import mark_validation_errors_as_inbound
+from event.step_chain import StepChain
+
+
+@mark_validation_errors_as_inbound
+def parse_path_params(data, cache) -> CpmProductPathParams:
+    event = APIGatewayProxyEvent(data[StepChain.INIT])
+    return CpmProductPathParams(**event.path_parameters)
+
+
+def read_cpm_product(data, cache) -> dict:
+    """
+    Read the CPM product from the DynamoDB table based on the product ID and team ID.
+    """
+    path_params: CpmProductPathParams = data[parse_path_params]
+
+    product_repo = CpmProductRepository(
+        table_name=cache["DYNAMODB_TABLE"], dynamodb_client=cache["DYNAMODB_CLIENT"]
+    )
+    cpm_product = product_repo.read(
+        product_id=path_params.product_id, product_team_id=path_params.product_team_id
+    )
+
+    return cpm_product.state()
```

Comment:
Does this work? I've had to add the state method on my ticket?

Reply:
Yep, but the method has been moved into the aggregate root (Joel pointed out that it's shared by devices as well for example)